pyrk/th_component.py

```diff
@@ -35,27 +42,68 @@ def __init__(self, name=None,
         :type timer: Timer object
         :param heatgen: is this component a heat generator (fuel)
         :type heatgen: bool
+        :param power_tot: power generated in this component
+        :type power_tot: float
+        :param sph: is this component a spherical component, spherical equations
+        for heatgen, conduction are different, post-processing is different too
+        :type sph: bool
+        :param ri: inner radius of the sph/annular component, ri=0 for sphere
+        :type ri: float
+        :param ro: outer radius of the sph/annular component, ro=radius for sphere
+        :type ro: float
         """
         self.name = name
-        self.vol = vol.to('meter**3')
-        validation.validate_ge("vol", vol, 0*units.meter**3)
+        self.vol = vol
+        self.mat = mat
```

Comment:
this is somewhat redundant with the next few lines. We should probably choose just one option. Either hold the material as an attribute of the component, or hold its values. I tend to want to hold the values, but I'll keep reading to see how you're using it. Either could be the right choice, but probably not both.